cmd/dra-example-controller/driver.go

```diff
@@ -94,9 +94,13 @@ func (d driver) GetClaimParameters(ctx context.Context, claim *resourcev1.Resour
 	}
 }
 
-func (d driver) Allocate(ctx context.Context, claim *resourcev1.ResourceClaim, claimParameters interface{}, class *resourcev1.ResourceClass, classParameters interface{}, selectedNode string) (*resourcev1.AllocationResult, error) {
+func (d driver) Allocate(ctx context.Context, claimAllocations []*controller.ClaimAllocation, selectedNode string) {
+
 	if selectedNode == "" {
-		return nil, fmt.Errorf("TODO: immediate allocations not yet supported")
+		for _, claim := range claimAllocations {
```

Comment:
This PR adds multiple for loops inside `Allocate` and as part of changes indention of several code lines.

I suspect a much smaller patch would be to implement `Allocate` with a for loop that calls an `allocate(ctx context.Context, claim *resourcev1.ResourceClaim, claimParameters interface{}, class *resourcev1.ResourceClass, classParameters interface{}, selectedNode string) (*resourcev1.AllocationResult, error)` method.

It's probably also going to be more readable.

Reply:
Something like this:
```
func (d driver) Allocate(ctx context.Context, claimAllocations []*controller.ClaimAllocation, selectedNode string) {
   for i := range claimAllocations {
       claimAllocations[i].Allocation, claimAllocations[i].Error = d.allocate(...)
   }
}

func (d driver) allocate(ctx context.Context, claim *resourcev1.ResourceClaim, claimParameters interface{}, class *resourcev1.ResourceClass, classParameters interface{}, selectedNode string) (*resourcev1.AllocationResult, error) {
  <same code as before>
}
```

Reply:
I understand that this then does some work multiple times. But I prefer readability over performance here.

Reply:
Done.